lib/sqs.js

```diff
@@ -233,9 +235,14 @@ module.exports = ({
         }
       }
 
-      let key = false;
+      let key = false
+        ,level = zlib.constants.Z_DEFAULT_COMPRESSION;
 
-      return compress(payload)
+      if (computeMessageSize(payload, attrs) < MAX_COMPRESS_MSG_SIZE) {
```

Comment:
oh yeah, i did have a slight nitpick here but its probably not a big deal, if we save off the message size:

```
const messageSize = computeMessageSize(payload, attrs)
```

We can avoid double-calling this later on, it's not a huge deal of work to compute but we do these operations a lot so... 🤷‍♂️

Reply:
I def. agree with this.  Feel dumb missing it. Updating.